coderd/workspaces_test.go

```diff
@@ -426,47 +426,441 @@ func TestWorkspace(t *testing.T) {
 
 	t.Run("TemplateVersionPreset", func(t *testing.T) {
 		t.Parallel()
-		client, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
-		user := coderdtest.CreateFirstUser(t, client)
-		authz := coderdtest.AssertRBAC(t, api, client)
-		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
-			Parse: echo.ParseComplete,
-			ProvisionPlan: []*proto.Response{{
-				Type: &proto.Response_Plan{
-					Plan: &proto.PlanComplete{
-						Presets: []*proto.Preset{{
-							Name: "test",
-						}},
+
+		testCases := []struct {
+			name                      string
+			presets                   []*proto.Preset
+			templateVersionParameters []*proto.RichParameter
+			selectedPresetIndex       int // -1 if no preset should be used
+		}{
+			{
+				name:                      "No Presets",
+				presets:                   []*proto.Preset{},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       -1,
+			},
+			{
+				name: "Single Preset - No Parameters",
+				presets: []*proto.Preset{{
+					Name: "test",
+				}},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       0,
+			},
+			{
+				name: "Single Preset - With Parameters But No Template Parameters",
+				presets: []*proto.Preset{{
+					Name: "test",
+					Parameters: []*proto.PresetParameter{
+						{Name: "param1", Value: "value1"},
+						{Name: "param2", Value: "value2"},
+					},
+				}},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       0,
+			},
+			{
+				name: "Single Preset - With Matching Parameters",
+				presets: []*proto.Preset{{
+					Name: "test",
+					Parameters: []*proto.PresetParameter{
+						{Name: "param1", Value: "value1"},
```

Comment:
Nit: use consts/vars instead of repeating the same strings; subtle typos can have you ripping your hair out.
In fact, if you use a var you can pass a pointer to an _actual_ preset definition, obviating the need for `selectedPresetIndex`.